src/parsers/pm_io/data.jl

```diff
@@ -1796,3 +1796,28 @@ function _make_multiconductor(data::Dict{String,Any}, conductors::Real)
         end
     end
 end
+
+
+"""
+Move gentype and genfuel fields to be subfields of gen
+"""
+function move_genfuel_and_gentype!(data::Dict{String,Any})
+    ngen = length(data["gen"])
+    
+    toplevkeys = ("genfuel", "gentype")
+    sublevkeys = ("fuel", "type")
+    for i in range(1, stop=2)
```

Comment:
I think this would be more clear and robust if you iterated over the keys rather than a range of the hard-coded current size.

Reply:
I've got two sets of matching keys. I'm not sure how to make that iteration work except to go by numerical position.

Reply:
At a minimum you could set "stop" to the length of the tuple instead of 2.